cmd/kubermatic-installer/cmd_quickstart.go

```diff
@@ -0,0 +1,355 @@
+	}
+	ctrlConfig, err := ctrlruntimeconfig.GetConfig()
+	if err != nil {
+		logger.Fatalf("failed to initialize runtime config: %v", err)
+	}
+
+	mgr, err := manager.New(ctrlConfig, manager.Options{
+		MetricsBindAddress:     "0",
+		HealthProbeBindAddress: "0",
+	})
+	if err != nil {
+		logger.Fatalf("failed to construct mgr: %v", err)
+	}
+
+	// start the manager in its own goroutine
+	appContext := context.Background()
+
+	go func() {
+		if err := mgr.Start(appContext); err != nil {
+			logger.Fatalf("Failed to start Kubernetes client manager: %v", err)
+		}
+	}()
+
+	// wait for caches to be synced
+	mgrSyncCtx, cancel := context.WithTimeout(appContext, 30*time.Second)
+	defer cancel()
+	if synced := mgr.GetCache().WaitForCacheSync(mgrSyncCtx); !synced {
+		logger.Fatal("Timed out while waiting for Kubernetes client caches to synchronize.")
+	}
+	return mgr.GetClient(), cancel
+}
+
+func ensureResource(kubeClient ctrlruntimeclient.Client, logger *logrus.Logger, o ctrlruntimeclient.Object) {
+	if err := kubeClient.Create(context.Background(), o); err != nil && !apierrors.IsAlreadyExists(err) {
+		logger.Fatal(err)
+	}
+}
+
+func installKubevirt(logger *logrus.Logger, dir string) {
+	logger.Info("Installing KubeVirt ...")
+	kubeconfig := filepath.Join(dir, "kube-config.yaml")
+	helmClient, err := helm.NewCLI(helmBin, kubeconfig, "", time.Minute, logger)
+	if err != nil {
+		logger.Fatalf("Failed to create helm client: %v", err)
+	}
+	err = helmClient.InstallChart("", "kubevirt", filepath.Join(helmChartDir, "quickstart-kubevirt"), "", nil, nil)
+	if err != nil {
+		logger.Fatalf("Failed to install KubeVirt Helm client: %v", err)
+	}
+}
+
+func installKubermatic(logger *logrus.Logger, dir string, kubeClient ctrlruntimeclient.Client) string {
+	ip := getLocalIP(logger)
+	kkpEndpoint := fmt.Sprintf(kkpEndpointTemplate, ip)
+	logger.Infof("Installing KKP at %v ...", ip) // TODO: prettify
+	valuesExamplePath := filepath.Join(dir, "values.example.yaml")
+	kubermaticExamplePath := filepath.Join(dir, "kubermatic.example.yaml")
+	valuesPath := filepath.Join(dir, "values.yaml")
+	kubermaticPath := filepath.Join(dir, "kubermatic.yaml")
+
+	_, uk, err := loadKubermaticConfiguration(kubermaticExamplePath)
+	if err != nil {
+		logger.Fatalf("failed to unmarshal example kubermatic.yaml: %v", err)
+	}
+	// TODO: make this idempotent (e.g. if generated yamls already exist, don't touch them)
+	uSetNestedField(logger, uk.Object, kkpEndpoint, "spec", "ingress", "domain")
+	uSetNestedField(logger, uk.Object, nil, "spec", "ingress", "certificateIssuer")
+	uSetNestedField(logger, uk.Object, fmt.Sprintf("http://%v/dex", kkpEndpoint), "spec", "auth", "tokenIssuer")
+	uSetNestedField(logger, uk.Object, randomString(32), "spec", "auth", "issuerClientSecret")
+	uSetNestedField(logger, uk.Object, randomString(32), "spec", "auth", "issuerCookieKey")
+	uSetNestedField(logger, uk.Object, randomString(32), "spec", "auth", "serviceAccountKey")
```

Comment:
I actually came to the conclusion any marshalling makes the UX suboptimal here. It's nice and easy from the programmer's perspective but then it re-arranges the document during marshalling which makes it hard to diff between the `kubermatic.example.yaml` and autoconfigured `kubermatic.yaml`. I would like to later propose pure text processing without marshal/unmarshal but imho not in scope for now.

Reply:
I'm wondering if users are actually interested in the generated configuration files here? Since they have a very narrow scope, I don't think users should really care about them. They are specifically not a good starting point for configuration.

The reason I'm bringing `yamled` up is because of the `kubermatic-installer init` command I'm working on over at https://github.com/embik/kubermatic/tree/interactive-init/pkg/install/init. I had the same problem of generating a configuration file and updating existing examples, and this is where I'm setting Helm values:

https://github.com/embik/kubermatic/blob/7c4c52f426fd81cb98761d6f10cff727236c96c7/pkg/install/init/generator/generator.go#L147

Reply:
For now, this should be okay. We should check it out again later so we don't end up reinventing the wheel though.